scanner/indexer/indexer_test.go

```diff
@@ -60,3 +68,49 @@ func Test_parseContainerImageURL(t *testing.T) {
 		})
 	}
 }
+
+func TestNewLibindex(t *testing.T) {
+	ctx := context.Background()
+	s := mockIndexerStore(t)
+	yamlData := `
+http_listen_addr: 127.0.0.1:9443
+grpc_listen_addr: 127.0.0.1:8443
+indexer:
+  enable: true
+  database:
+    conn_string: "host=/var/run/postgresql"
+    password_file: ""
+  get_layer_timeout: 1m
+  repository_to_cpe_url: https://storage.googleapis.com/scanner-v4-test/redhat-repository-mappings/repository-to-cpe.json
+  name_to_cpe_url: https://storage.googleapis.com/scanner-v4-test/redhat-repository-mappings/container-name-repos-map.json
+matcher:
+  enable: true
+  database:
+    conn_string: "host=/var/run/postgresql"
+    password_file: ""
+mtls:
+  certs_dir: ""
+log_level: info
+`
+	reader := strings.NewReader(yamlData)
+	ic, err := loadIndexerConfig(reader)
+	require.NoError(t, err)
+	_, err = newLibindex(zlog.Test(ctx, t), ic, s, nil)
+	require.NoError(t, err)
```

Comment:
Up to you, somewhat outside the scope of `newLibindex`, but could be worth validating that the various options were set, something like:

```suggestion
	reader := strings.NewReader(yamlData)
	ic, err := loadIndexerConfig(reader)
	require.NoError(t, err)
	indexer, err := newLibindex(zlog.Test(ctx, t), ic, s, nil)
	require.NoError(t, err)
	assert.NotNil(t, indexer.Options.ScannerConfig.Repo["rhel-repository-scanner"])
	assert.NotNil(t, indexer.Options.ScannerConfig.Package["rhel_containerscanner"])
```